lamar/tasks/mapping.py

```diff
@@ -94,6 +95,19 @@ def __init__(self, config, outputs, capture, session_id,
             self.name2key[image.name]: image.image_id
             for image in self.reconstruction.images.values()
         }
+        # We cache the 3D points for mapping images to avoid parsing them from
```

Comment:
Actually in https://github.com/microsoft/lamar-benchmark/pull/62 I already had 100x speedup in pose estimation by only caching the image-to-point3D ID mapping, would this be sufficient? fetching the xyz seems already pretty fast and caching it is more expensive memory-wise.

Reply:
I'll take a look there - maybe that's sufficient since I agree the locks complicate things.

To be honest, the fastest I managed to get our pose estimation was with single-threaded + caching. I didn't investigate too much where it's coming from though.